sensor/tests/helper/helper.go

```diff
@@ -264,7 +275,21 @@ func (c *TestContext) createTestNs(ctx context.Context, t *testing.T, name strin
 	})
 	nsObj := v1.Namespace{}
 	nsObj.Name = name
-	if err := c.r.Create(ctx, &nsObj); err != nil {
+
+	err := k8swait.ExponentialBackoff(getDefaultBackoff(), func() (bool, error) {
+		if err := c.r.Create(ctx, &nsObj); err != nil {
+			t.Logf("failed to create %q %q: %v\n", nsObj.Kind, nsObj.Name, err)
+			return false, nil
```

Comment:
Did you maybe mean
```suggestion
			return false, err
```
?
Otherwise the `err` will be always nil.

Reply:
My understanding from the documentation is that the retry will be stopped on returning `true` or any `error`. 

Docs here: https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#ExponentialBackoff

So, if we return an error on this line, it will automatically stop retrying (we will have only one try). One way to keep the error is to create a variable outside of the closure (i.e. `lastErr`) and keep it tracked.